### ISSUE TITLE
[iclr_tadv.py] Question about the style loss and gram matrix.

### ISSUE THREAD

**Issue by xiangyh9988** [AI-secure/Big-but-Invisible-Adversarial-Attack#1]
Hi, thanks for your interesting work! I got a little confused about the calculation about the style loss, especially `class CrossGramMatrix()` (line 124) and `layer_losses = [weights[a] * loss_fns[a](A,B, Crossstyle_targets[a])/CrossGramMatrix()(A,B).std()  for a,(A,B) in enumerate(Cross_out)]`  (line 285) in the code. My questions are:

1. In `CorssGramMatrix`, why is the G calculated by torch.bmm divided by h1*w1 then? I'm not very clear about the Gram Matrix before and just get some knowledge about it from `Gatys et al. 2015` and `Yeh et al. 2020` from the reference of your paper. And in the paper, i didn't find the division in Eq.(4). Maybe I didn't get a clear definition about Gram Matrix? If so, please tell me and I will check more about that.

2. In line 285, the standard deviation of the Gram Matrix was divided, while $std(G_{ij}^{m,n}(I_v))$ was divided in Eq.(4). It seems that $G_{ij}^{m,n}(I_v))$ is not a matrix, so how to get its STD? Is it a typo? And, is this division the additional constraint on the variation in the gram matrices?

By the way, could you provide files or their links mentioned in `iclr_tadv.py`? 
Apologize for my silly questions and thank your in advance!

**Comment by SJHNJU** [comment]
Have you successfully run tadv.py by now? I meet the same question

**Comment by xiangyh9988** [comment]
> Have you successfully run tadv.py by now? I meet the same question

No, I haven't. I didn't follow this work from then and apologize that I can't provide you with some help.
